src/utils.ts

```diff
@@ -409,7 +409,7 @@ export async function logDuration<T>(logger: BaseLogger, name: string, block: ()
  * Promise/async version of {@link setImmediate}.
  */
 export function immediate(): Promise<void> {
-    return new Promise(setImmediate);
+    return new Promise((resolve) => setTimeout(resolve));
```

Comment:
Given this is only used in one place, can we just kill it off (possibly replacing it with a call to `sleep(0)`) ?

Otherwise, it feels like the documentation could do with updating to explain what it does without reference to a heavily deprecated function.

Reply:
Not without a breaking change, given its in an export which other packages use.

Reply:
UGH. this is not something the js-sdk ought to be exporting.

Let's improve the documentation then, and/or deprecate it?